meson.build

```diff
@@ -2478,6 +2569,61 @@ foreach tuple : [['myhostname', 'ENABLE_NSS_MYHOSTNAME'],
         endif
 endforeach
 
+foreach dict : modules
+        name = dict.get('name')
+        is_nss = name.startswith('nss_')
+        is_pam = name.startswith('pam_')
+
+        build = true
+        foreach cond : dict.get('conditions', [])
+                if conf.get(cond) != 1
+                        build = false
+                        break
+                endif
+        endforeach
+        if not build
+                continue
+        endif
+
+        kwargs = {}
+        foreach key, val : dict
+                if key in ['name', 'conditions']
+                        continue
+                endif
+                kwargs += { key : val }
+        endforeach
+
+        sym = meson.current_source_dir() / '@0@'.format(dict.get('link_depends')[0])
+
+        # Note that we link NSS modules with '-z nodelete' so that mempools never get orphaned
+        link_args = ['-shared',
+                     '-Wl,--version-script=' + sym]
+        if is_nss
+                link_args += ['-Wl,-z,nodelete']
```

Comment:
Not that matters, but `-Wl,` should not be needed as the compiler understands `-z`...